### PR TITLE
Bug 1695654: Add a method to fetch dataset labels from dryrun service results

### PR DESCRIPTION
**Pull request by jklukas**
Depends on https://github.com/mozilla-services/cloudops-infra/pull/2916 ; marking as draft until that's merged.

### CODE REVIEW

File: bigquery_etl/dryrun.py [hunk 1]
```diff
@@ -328,6 +328,24 @@ def get_schema(self):
...
+        if self.sqlfile in SKIP:
+            print(f"\t...Ignoring dryrun results for {self.sqlfile}")
+            return {}
+
+        if (
+            self.dry_run_result
+            and self.dry_run_result["valid"]
+            and "datasetLabels" in self.dry_run_result
+        ):
+            return self.dry_run_result["dataset_labels"]
```
Comment (scholtzan) [inline bigquery_etl/dryrun.py:345]:
```suggestion
            return self.dry_run_result["datasetLabels"]
```

**Review by scholtzan — APPROVED** [review]
I think the dryrun cloud function changes got merged

**Comment by jklukas** [comment]
Some tests are failing, and I think the problem is that the tests are using a path that contains a placeholder dataset name, so the getDataset call in the dry run fails, and now if fails the whole dry run. Example error message:

> [{'code': 404, 'errors': [{'message': 'Not found: Dataset moz-fx-data-shared-prod:test_get_referenced_tables0', 'domain': 'global', 'reason': 'notFound'}], 'response': {'headers': {'alt-svc': 'h3-29=":443"; ma=2592000,h3-T051=":443"; ma=2592000,h3-Q050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"', 'cache-control': 'private', 'content-encoding': 'gzip', 'content-type': 'application/json; charset=UTF-8', 'date': 'Mon, 01 Mar 2021 20:44:19 GMT', 'server': 'ESF', 'transfer-encoding': 'chunked', 'vary': 'Origin, X-Origin, Referer', 'x-content-type-options': 'nosniff', 'x-frame-options': 'SAMEORIGIN', 'x-xss-protection': '0'}}, 'message': 'Not found: Dataset moz-fx-data-shared-prod:test_get_referenced_tables0'}]

So in the dry run service, I think we need to change the logic to still be able to return a valid result even if the getDataset part fails. Or we need to refactor tests to accommodate this.

**Comment by scholtzan** [comment]
Updating the dry run service seems like a good option. I can push an update for this.

**Comment by jklukas** [comment]
It also looks pretty straight-forward to update tests here to include a realistic directory structure, so I'm making an effort at that.

**Comment by scholtzan** [comment]
I also pushed a fix here for the dry run function: https://github.com/mozilla-services/cloudops-infra/pull/2918

**Pull request merged**